frontend-react/src/components/Spinner.tsx

```diff
@@ -1,32 +1,45 @@
 import { Oval } from "react-loader-spinner";
+import { ReactNode } from "react";
 
 interface SpinnerProps {
     size?: "default" | "fullpage" | "insidebutton";
     display?: boolean; // allows easier dynamic show/hide during load
+    message?: ReactNode;
 }
 
-function Spinner({ size = "default", display = true }: SpinnerProps) {
+function Spinner({
+    size = "default",
+    display = true,
+    message = "",
+}: SpinnerProps) {
     // map prop to css className
     const sizeClassName = {
         default: "grid-container rs-spinner-default",
         fullpage: "grid-container rs-spinner-default rs-spinner-fullpage",
         insidebutton: "rs-spinner-default rs-spinner-tiny",
     }[size];
     return (
-        <div
-            hidden={!display}
-            className={sizeClassName}
-            data-testid="rs-spinner"
-        >
-            <Oval
-                height={60}
-                width={60}
-                color="#2378c3"
-                secondaryColor="#aacdec"
-                strokeWidth={8}
-                strokeWidthSecondary={8}
-                ariaLabel="loading-indicator"
-            />
+        <div>
+            <div
```

Comment:
The styles of this are off:
<img width="1015" alt="Screenshot 2023-03-24 at 14 33 04" src="https://user-images.githubusercontent.com/2421042/227610833-f66428fb-24a8-4971-965e-7e91148397b2.png">

Two things are causing this:
- `grid-container` is applied to the spinner container, which means that it'll take up the full width
- The message is wrapped in a `span`, which is inline and _won't_ take up the full width

Reply:
(There should also probably be some spacing between the tip and the spinner)